src/transports/quic.rs

```diff
@@ -12,25 +12,41 @@ use mio::{net::UdpSocket as MioUdpSocket, Events, Interest, Poll, Token, Waker};
 use parking_lot::RwLock;
 
 use crate::{
-    error::PeerNetError,
+    error::{PeerNetError, PeerNetResult},
     handlers::MessageHandlers,
     network_manager::{FallbackFunction, HandshakeFunction, SharedActiveConnections},
     peer::{new_peer, HandshakeHandler},
     peer_id::PeerId,
-    transports::Endpoint,
+    transports::{Endpoint, TransportErrorType},
 };
 
 use super::Transport;
 
 const NEW_PACKET_SERVER: Token = Token(0);
 const STOP_LISTENER: Token = Token(10);
 
+#[derive(Debug)]
+pub enum QuicError {
+    InitListener,
+    StopListener,
+    SocketConfig,
+    QuicheConfig,
+    ConnectionError,
+    InternalFail,
+}
+
+impl QuicError {
+    fn wrap(self) -> PeerNetError {
```

Comment:
I'm not a big fan of this. Is that possible to use the `From` impl for `PeerNetError` from `TransportErrorType` instead of having an ad hoc function ? Maybe a personal opinion

Reply:
Yes but as the type isn't known, that would require to have `QuicError::ErrorType.into::<PeerNetError::TransportError>()` so having a `wrap` function is much less verbose and allows for a better typing match here.